lib/Classifiers/Video/MovinetClassifier.php

```diff
@@ -7,117 +7,61 @@
 
 namespace OCA\Recognize\Classifiers\Video;
 
+use OCA\Recognize\Classifiers\Classifier;
+use OCA\Recognize\Db\Video;
+use OCA\Recognize\Db\VideoMapper;
 use OCA\Recognize\Service\Logger;
 use OCA\Recognize\Service\TagManager;
-use OCP\Files\File;
+use OCP\DB\Exception;
+use OCP\Files\IRootFolder;
 use OCP\IConfig;
-use OCP\Files\InvalidPathException;
-use OCP\Files\NotFoundException;
-use Psr\Log\LoggerInterface;
-use Symfony\Component\Process\Exception\ProcessTimedOutException;
-use Symfony\Component\Process\Exception\RuntimeException;
-use Symfony\Component\Process\Process;
 
-class MovinetClassifier {
+class MovinetClassifier extends Classifier {
 	public const VIDEO_TIMEOUT = 480; // seconds
 	public const MODEL_DOWNLOAD_TIMEOUT = 180; // seconds
+	public const MODEL_NAME = 'movinet';
 
-	/**
-	 * @var LoggerInterface
-	 */
-	private $logger;
-
+	private Logger $logger;
 	private TagManager $tagManager;
-
 	private IConfig $config;
+	private IRootFolder $rootFolder;
+	private VideoMapper $videoMapper;
 
-	public function __construct(Logger $logger, IConfig $config, TagManager $tagManager) {
+	public function __construct(Logger $logger, IConfig $config, TagManager $tagManager, IRootFolder $rootFolder, VideoMapper $videoMapper) {
+		parent::__construct($logger, $config);
 		$this->logger = $logger;
 		$this->config = $config;
 		$this->tagManager = $tagManager;
+		$this->rootFolder = $rootFolder;
+		$this->videoMapper = $videoMapper;
 	}
-
 	/**
-	 * @param File[] $files
+	 * @param \OCA\Recognize\Db\Video[] $videos
 	 * @return void
 	 * @throws \OCP\Files\NotFoundException
-	 * @throws \OCP\Files\InvalidPathException
 	 */
-	public function classify(array $files): void {
-		$paths = array_map(function ($file) {
+	public function classify(array $videos): void {
+		$paths = array_map(function (Video $image) : string {
+			$file = $this->rootFolder->getById($image->getFileId())[0];
```

Comment:
This will not throw but return an empty array, so something that would need some error handling.

Reply:
(for all classifiers basically)

Reply:
Ah, nice catch :)